integration-tests/sql/src/main/java/org/apache/camel/quarkus/component/sql/it/SqlRoutes.java

```diff
@@ -0,0 +1,148 @@
+        //db has to be initialized before routes are started
+        sqlDbInitializer.initDb();
+
+        from("sql:select * from projects where processed = false order by id?initialDelay=0&delay=50&consumer.onConsume=update projects set processed = true where id = :#id")
+                .id("consumerRoute").autoStartup(false)
+                .process(e -> results.get("consumerRoute").add(e.getMessage().getBody(Map.class)));
+
+        from("sql:classpath:sql/selectProjects.sql?initialDelay=0&delay=50&consumer.onConsume=update projects set processed = true")
+                .id("consumerClasspathRoute").autoStartup(false)
+                .process(e -> results.get("consumerClasspathRoute").add(e.getMessage().getBody(Map.class)));
+
+        Path tmpFile = createTmpFileFrom("sql/selectProjects.sql");
+        from("sql:file:" + tmpFile
+                + "?initialDelay=0&delay=50&consumer.onConsume=update projects set processed = true")
+                        .id("consumerFileRoute").autoStartup(false)
+                        .process(e -> results.get("consumerFileRoute").add(e.getMessage().getBody(Map.class)));
+
+        from("direct:transacted")
+                .transacted("PROPAGATION_REQUIRED")
+                .to("sql:overriddenByTheHeader")
+                .process(e -> {
+                    if (e.getIn().getHeader("rollback", boolean.class)) {
+                        throw new Exception("forced Exception");
+                    }
+                });
+
+        // Idempotent Repository
+        JdbcMessageIdRepository repo = new JdbcMessageIdRepository(dataSource, "idempotentRepo");
+        from("direct:idempotent")
+                .idempotentConsumer(header("messageId"), repo)
+                .process(e -> results.get("idempotentRoute").add(e.getMessage().getBody(String.class)));
+
+        //aggregation repository
+        JdbcAggregationRepository aggregationRepo = new JdbcAggregationRepository(
+                new JtaTransactionManager(userTransaction, tm), "aggregation", dataSource);
+        from("direct:aggregation")
+                .aggregate(header("messageId"), new MyAggregationStrategy())
+                // use our created jdbc repo as aggregation repository
+                .completionSize(4).aggregationRepository(aggregationRepo)
+                .process(e -> results.get("aggregationRoute").add(e.getMessage().getBody(String.class)));
+
+    }
+
+    private Path createTmpFileFrom(String file) throws IOException {
+        File tmpFile = File.createTempFile("selectProjects-", ".sql");
+        tmpFile.deleteOnExit();
+        try (InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream(file)) { //todo quick workaround
+
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            int c;
+            while ((c = is.read()) >= 0) {
+                out.write(c);
+            }
+            new FileOutputStream(tmpFile).write(out.toByteArray());
```

Comment:
Not closing a FileOutputStream may lead to surprising reads and perhaps also file being locked on Windows. `java.nio.file.Files.write(Path, byte[], OpenOption...)` might be a good replacement for this line, or even better the whole try block could be replaced by  `java.nio.file.Files.copy(InputStream, Path, CopyOption...)`

Reply:
My fault, I'll fix it.

Reply:
fixed